projects/igniteui-angular/src/lib/directives/radio/radio-group.directive.ts

```diff
@@ -301,8 +306,10 @@ export class IgxRadioGroupDirective implements AfterContentInit, ControlValueAcc
         // the OnInit of the NgModel occurs after the OnInit of this class.
         this._isInitialized = true;
 
-        setTimeout(() => {
-            this._initRadioButtons();
+        this.radioButtons.changes.pipe(startWith(0), takeUntil(this.destroy$)).subscribe(() => {
+            this.queryChange$.next();
+            this.queryChange$.complete();
```

Comment:
I am not sure what the purpose of the `queryChange$` is in here. When the radio button change it gets completed immediately and not used again.

Reply:
We discussed this in a meeting that it might not be needed to have this Subject at all. But the idea here is that if the same button instance is kept after an ngFor change, it will already have a subscription from a previous `changes` handler and that needs to be removed before subscribing it again.

You can try testing this for memory leaks without the queryChange handler too and if it's good, I'll remove the whole thing.

Reply:
I see, still though the `queryChange$` actually doesn't have any effect in the `button.change.pipe` from what I can see. In the following scenario `_selectedRadioButtonChanged` get triggered multiple times, as many as the added new radios to the `ngFor`. How to reproduce it:

1. Push one by one values to a list used in `ngFor` inside the igxRadioGroup.
2. Click on one of the old radios to select it
3. Observe how many times the `_selectedRadioButtonChanged` triggers.

I think the `merge` combined with `takeUntil` in this scenario acts as `takeUntil` all the Observables in the merge have completed.
I would suggest another approach like using a `IterableDiffer ` to determine the added radios from the `this.radioButtons.changes` by comparing `ids` and the calling `_initRadioButtons` for a specified list of radios or something like that.